cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -399,6 +498,13 @@ TYPED_TEST_SUITE(TestMeanKernelNumeric, NumericArrowTypes);
 TYPED_TEST(TestMeanKernelNumeric, SimpleMean) {
   using ScalarType = typename TypeTraits<DoubleType>::ScalarType;
 
+  const ScalarAggregateOptions& options =
+      ScalarAggregateOptions(ScalarAggregateOptions::SKIPNA, 0);
+
+  ValidateMean<TypeParam>("[]", Datum(std::make_shared<ScalarType>(0)), options);
```

Comment:
Just to make sure I am reading this one correctly, is it basically testing `mean([], min_count=0) == 0` ?

I am not sure that should give 0 as result? I would have expected either NaN or Null. 
Basically mean of empty is dividing 0/0, which is NaN.

Reply:
Agreed. I think I've put it there to have it documented while I was changing other things.

Reply:
It seems you changed it to return null now? Should it rather return NaN? (I assume it could be consistent with a manual mean from sum/count with `min_count=0`)